src/deploy/functions/deploy.ts

```diff
@@ -0,0 +1,53 @@
+import * as gcp from "../../gcp";
+import * as clc from "cli-color";
+import { setGracefulCleanup } from "tmp";
+import { logBullet, logSuccess, logWarning } from "../../utils";
+import * as prepareFunctionsUpload from "../../prepareFunctionsUpload";
+
+const GCP_REGION = gcp.cloudfunctions.DEFAULT_REGION;
+
+setGracefulCleanup();
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+async function uploadSource(
+  context: { projectId: string; uploadUrl?: string },
+  source: any
+): Promise<void> {
+  const uploadUrl = await gcp.cloudfunctions.generateUploadUrl(context.projectId, GCP_REGION);
+  context.uploadUrl = uploadUrl;
+  const apiUploadUrl = uploadUrl.replace("https://storage.googleapis.com", "");
+  await gcp.storage.upload(source, apiUploadUrl);
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+module.exports = async function(context: any, options: any, payload: any): Promise<any> {
```

Comment:
Didn't even notice this during the conversion. Done, but it makes some other things a little ugly because of how these are structured and I'm not prepared to touch `prepare.js` or `release.js` yet and incur their lint explosion.